navigation/AppNavigation.js

```diff
@@ -1,39 +1,73 @@
+/* eslint-disable react/prop-types */
 import React from 'react';
-import { createStackNavigator, createAppContainer } from 'react-navigation';
+import { createStackNavigator, createBottomTabNavigator } from 'react-navigation';
 
-import { white } from '../scss/colors';
+import { white, lightBlue, gray } from '../scss/colors';
+import IconComponent from '../app/components/baseComponents/IconComponent';
 import LibraryContainer from '../app/screens/Library/index';
 import BookDetailContainer from '../app/screens/BookDetail/index';
+import WishlistContainer from '../app/screens/WishList';
+import { ROUTES_TITLES } from '../constants/routes';
 
 import ImageHeader from './components/ImageHeader';
 import BackButtonHeader from './components/BackButtonHeader';
 
-const AppNavigator = createStackNavigator(
+const LibraryStack = createStackNavigator(
   {
-    library: {
-      screen: LibraryContainer,
-      navigationOptions: {
-        title: 'Library'
-      }
-    },
-    bookDetail: {
-      screen: BookDetailContainer,
-      navigationOptions: {
-        title: 'Book Detail'
-      }
-    }
+    library: { screen: LibraryContainer },
+    bookDetail: { screen: BookDetailContainer }
   },
   {
     initialRouteName: 'library',
-    defaultNavigationOptions: {
+    defaultNavigationOptions: ({ navigation }) => ({
       headerBackground: <ImageHeader />,
       headerTitleStyle: {
         fontWeight: 'bold',
         color: white
       },
-      headerBackImage: BackButtonHeader
+      headerBackImage: BackButtonHeader,
+      headerTitle: ROUTES_TITLES[navigation.state.routeName]
+    })
+  }
+);
+
+const WishlistStack = createStackNavigator(
+  {
+    wishlist: { screen: WishlistContainer }
+  },
+  {
+    initialRouteName: 'wishlist',
+    defaultNavigationOptions: ({ navigation }) => ({
+      headerBackground: <ImageHeader />,
+      headerTitleStyle: {
+        fontWeight: 'bold',
+        color: white
+      },
+      headerBackImage: BackButtonHeader,
+      headerTitle: ROUTES_TITLES[navigation.state.routeName]
```

Comment:
all navigation options should be in a different file than where you create your navigator